### PR TITLE
Upgrade xorm

### PR DESCRIPTION
**Pull request by nanguanlin6**
Related to https://gitea.com/xorm/xorm/pulls/2341

### CODE REVIEW

**Review by wolfogre — APPROVED** [review]

**Comment by lng2020** [comment]
Need to backport because PR #27609 is backported.

**Review by delvh — APPROVED** [review]

**Comment by lng2020** [comment]
CI failure is related.  Seems like inappropriate usage of `group by`
ci_failure

**Review by lunny — APPROVED** [review]

**Comment by lng2020** [comment]
CI failure is related.
the latest version of xorm changes the implementation of `limit` when the database is mssql.
So the SQL query gets error when there is no `orderby`
```sql
SELECT count(*) FROM [commit_status] WHERE (repo_id = ?) AND (sha = ?) OFFSET 0 ROWS FETCH NEXT 30 ROWS ONLY
```
see https://gitea.com/xorm/xorm/pulls/2188 for more detail

**Comment by lunny** [comment]
> CI failure is related. the latest version of xorm changes the implementation of `limit` when the database is mssql. So the SQL query gets error when there is no `orderby`
> 
> ```sql
> SELECT count(*) FROM [commit_status] WHERE (repo_id = ?) AND (sha = ?) OFFSET 0 ROWS FETCH NEXT 30 ROWS ONLY
> ```
> 
> see https://gitea.com/xorm/xorm/pulls/2188 for more detail

Looks like it's xorm's responsibility to fix the problem.

**Comment by lng2020** [comment]
> Looks like it's xorm's responsibility to fix the problem.

Yes. That PR seems to forget to add `orderby 1` in some cases. Though manually adding `orderby 1` in Gitea probably will solve the problem, it's better for xorm to implement that.

File: models/unittest/unit_tests.go [hunk 1]
```diff
@@ -96,7 +96,15 @@ func AssertExistsAndLoadMap(t assert.TestingT, table string, conditions ...any)
 // GetCount get the count of a bean
 func GetCount(t assert.TestingT, bean any, conditions ...any) int {
 	e := db.GetEngine(db.DefaultContext)
-	count, err := whereOrderConditions(e, conditions).Count(bean)
+	for _, condition := range conditions {
```
Comment (nanguanlin6) [inline models/unittest/unit_tests.go:99]: If the `limit` implementation of xorm is not fixed, this code will run into a bug too. But in our cases, we just don't use `limit` when writing test cases so the bug is not triggered.

**Comment by lng2020** [comment]
CI is passing. Merge this or wait for xorm?
Some noticeable changes:
1.  When using the function `GetCount` for test usage, avoid using the ` OrderBy id` clause since `id` is not selected.
2.  Manually add `OrderBy 1` when there is no `OrderBy`.

**Comment by delvh** [comment]
If we want to backport this PR, we should merge it soon.
I want to prepare 1.21.0-rc2.

**Comment by lunny** [comment]
We haven't used counting with limit yet in production codes, so I'm OK to merge this and I'm also sending https://gitea.com/xorm/xorm/pulls/2342 to try to fix it.

**Pull request merged**